File: MovieCatalogueV2/app/src/main/java/com/google/fdp/moviecataloguev2/SettingActivity.java
package com.google.fdp.moviecataloguev2;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.CompoundButton;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SwitchCompat;

import com.google.fdp.moviecataloguev2.alarm.DailyReminderManager;
import com.google.fdp.moviecataloguev2.alarm.ReleaseReminderManager;

public class SettingActivity extends AppCompatActivity {

    private static String RELEASE_REMINDER = "release_reminder";
    private static String DAILY_REMINDER = "daily_reminder";

    DailyReminderManager mDailyReminderManager;
    ReleaseReminderManager mReleaseReminderManager;

    Context mContext;

    SwitchCompat scReleaseReminder;
    SwitchCompat scDailyReminder;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);

        mContext = this;
        scReleaseReminder = findViewById(R.id.scReleaseReminder);
        scDailyReminder = findViewById(R.id.scDailyReminder);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
            actionBar.setTitle(getString(R.string.reminder_setting));
        }

        mDailyReminderManager = new DailyReminderManager();
        mReleaseReminderManager = new ReleaseReminderManager();

        scReleaseReminder.setChecked(checkReminder(RELEASE_REMINDER));
        scDailyReminder.setChecked(checkReminder(DAILY_REMINDER));

        scReleaseReminder.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
                if (isChecked) {
                    mReleaseReminderManager.setRepeatingAlarm(mContext, "08:00");
                } else {
                    mReleaseReminderManager.cancelAlarm(mContext);
                }
            }
        });


        scDailyReminder.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
                if (isChecked) {
                    mDailyReminderManager.setRepeatingAlarm(mContext, getString(R.string.daily_notification_title),
                            getString(R.string.daily_notification_message), "07:00");
                } else {
                    mDailyReminderManager.cancelAlarm(mContext);
                }
            }
        });

    }

    private void saveReminder(String key, Boolean value) {
        SharedPreferences sharedPreferences = getPreferences(Context.MODE_PRIVATE);
        if (sharedPreferences != null) {
            sharedPreferences.edit().putBoolean(key, value).apply();
        }
    }

    private Boolean checkReminder(String key) {
        SharedPreferences sharedPreferences = getPreferences(Context.MODE_PRIVATE);
        if (sharedPreferences != null) {
            return sharedPreferences.getBoolean(key, false);
        }
        return false;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: MovieCatalog 3/app/src/main/java/com/alfianh/moviecatalog/SettingActivity.kt
package com.alfianh.moviecatalog

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import com.alfianh.moviecatalog.alarm.DailyReminderManager
import com.alfianh.moviecatalog.alarm.ReleaseReminderManager
import kotlinx.android.synthetic.main.activity_setting.*

class SettingActivity : AppCompatActivity() {

  private lateinit var mDailyReminderManager: DailyReminderManager
  private lateinit var mReleaseReminderManager: ReleaseReminderManager

  companion object {
    const val RELEASE_REMINDER = "release_reminder"
    const val DAILY_REMINDER = "daily_reminder"
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_setting)
    supportActionBar?.setDisplayShowHomeEnabled(true)
    supportActionBar?.setDisplayHomeAsUpEnabled(true)
    title = getString(R.string.reminder_setting)
    mDailyReminderManager = DailyReminderManager()
    mReleaseReminderManager = ReleaseReminderManager()
    scReleaseReminder.isChecked = checkReminder(RELEASE_REMINDER)
    scDailyReminder.isChecked = checkReminder(DAILY_REMINDER)
    scReleaseReminder.setOnCheckedChangeListener { _, isChecked ->
      saveReminder(RELEASE_REMINDER, isChecked)
      if (isChecked) {
        mReleaseReminderManager.setRepeatingAlarm(this, "08:00")
      } else {
        mReleaseReminderManager.cancelAlarm(this)
      }
    }
    scDailyReminder.setOnCheckedChangeListener { _, isChecked ->
      saveReminder(DAILY_REMINDER, isChecked)
      if (isChecked) {
        mDailyReminderManager.setRepeatingAlarm(this, getString(R.string.daily_notification_title),
            getString(R.string.daily_notification_message), "07:00")
      } else {
        mDailyReminderManager.cancelAlarm(this)
      }
    }
  }

  private fun saveReminder(key: String, value: Boolean) {
    val sharedPref = getPreferences(Context.MODE_PRIVATE) ?: return
    with(sharedPref.edit()) {
      putBoolean(key, value)
      commit()
    }
  }

  private fun checkReminder(key: String): Boolean = getPreferences(Context.MODE_PRIVATE)?.let {
    it.getBoolean(key, false)
  } ?: false

  override fun onOptionsItemSelected(item: MenuItem?): Boolean {
    when (item?.itemId) {
      android.R.id.home -> {
        finish()
      }
    }
    return super.onOptionsItemSelected(item)
  }
}
